docs/source/integrations/triton.rst

```diff
@@ -1,3 +1,599 @@
+
+    - ``get_model_config``
+    - ``get_model_metadata``
+    - ``get_model_repository_index``
+    - ``is_model_ready``
+    - ``is_server_live``
+    - ``is_server_ready``
+    - ``load_model``
+    - ``unload_model``
+    - ``infer``
+    - ``stream_infer``
+
+    The following advanced client APIs are also supported:
+
+    - ``get_cuda_shared_memory_status``
+    - ``get_inference_statistics``
+    - ``get_log_settings``
+    - ``get_server_metadata``
+    - ``get_system_shared_memory_status``
+    - ``get_trace_settings``
+    - ``register_cuda_shared_memory``
+    - ``register_system_shared_memory``
+    - ``unregister_cuda_shared_memory``
+    - ``unregister_system_shared_memory``
+    - ``update_log_settings``
+    - ``update_trace_settings``
+
+.. epigraph::
+   :bdg-primary:`Important:` All of the client APIs are asynchronous. To use them, make sure to use it under an async ``@svc.api``. See :ref:`concepts/service:Sync vs Async APIs`
+
+   .. code-block:: python
+      :caption: `service.py`
+
+      @svc.api(input=bentoml.io.Text.from_sample("onnx_mnist"), output=bentoml.io.JSON())
+      async def unload_model(input_model: str):
+          await triton_runner.unload_model(input_model)
+          return {"unloaded": input_model}
+
+Packaging BentoService with Triton Inference Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To build your BentoService with Triton Inference Server, add the following to your ``bentofile.yaml``
+or use :ref:`reference/core:bentoml.bentos.build`:
+
+.. tab-set::
+
+   .. tab-item:: Building with ``bentofile.yaml``
+
+      .. literalinclude:: ./snippets/triton/bentofile.yaml
+         :language: yaml
+         :caption: `bentofile.yaml`
+
+      Building this Bento with :ref:`bentoml build <reference/cli:build>`:
+
+      .. code-block:: bash
+
+         $ bentoml build
+
+   .. tab-item:: Building with ``bentoml.bentos.build``
+
+      .. literalinclude:: ./snippets/triton/build_bento.py
+         :language: python
+         :caption: `build_bento.py`
+
+Notice that we are using ``nvcr.io/nvidia/tritonserver:22.12-py3`` as our base image. This can be substituted with any other
+custom base image that has ``tritonserver`` binary available. See Triton's documentation [#triton_build]_ to learn more about building/composing custom Triton image.
+
+.. epigraph::
+    :bdg-primary:`Important:` The provided Triton image from NVIDIA includes Python 3.8. Therefore, if you are developing your Bento
+    with any other Python version, make sure that your ``service.py`` is compatible with Python 3.8.
+
+Serving BentoService with Triton Inference Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After containerizing with :ref:`bentoml containerize <reference/cli:containerize>`, :ref:`serve <reference/cli:serve>`
+command now takes in additional ``--triton-options`` argument to pass options for ``tritonserver``:
```

Comment:
Will options defined in `--triton--options` apply to all triton runners?

Reply:
yes. Currently

Reply:
Let's add a note for this behavior.

Reply:
As discussed, these CLI options may be better as runner options, provided through `with_options()`. Aaron is testing out the feasibility. We will discuss again.